[harfbuzz] Harfbuzz with graphite2 feature generates references to dynamic imports when linking statically

### Is your feature request related to a problem? Please describe.

After building `harfbuzz[graphite2]` with static library and crt linkage, then trying to use it in a project, I get linking errors like the following:
```
harfbuzz.lib(hb-graphite2.cc.obj) : error LNK2019: unresolved external symbol __imp_gr_face_featureval_for_lang referenced in function _hb_graphite2_shape
harfbuzz.lib(hb-graphite2.cc.obj) : error LNK2019: unresolved external symbol __imp_gr_face_find_fref referenced in function _hb_graphite2_shape
```

As far as I can tell, this is caused by not setting `GRAPHITE2_STATIC` even though the crate is actually building it statically.

### Proposed solution

When static libraries are enabled, set `GRAPHITE2_STATIC` for harfbuzz when the `graphite2` feature is active

### Describe alternatives you've considered

Trying to set it myself - as far as I can tell, this requires patching stuff awkwardly. I'm not a master of vcpkg, so please tell me if there's a better way to handle this.

### Additional context

OS: Windows 10

Using custom triplet:
```
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE static)
set(VCPKG_LIBRARY_LINKAGE static)
set(VCPKG_BUILD_TYPE release)
```

--- a/ports/graphite2/portfile.cmake
+++ b/ports/graphite2/portfile.cmake
@@ -16,12 +16,15 @@ vcpkg_cmake_configure(
 vcpkg_cmake_install()
 vcpkg_copy_pdbs()
 vcpkg_cmake_config_fixup()
+vcpkg_fixup_pkgconfig()
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/graphite2/Types.h" "defined GRAPHITE2_STATIC" "1")
+endif()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 file(REMOVE "${CURRENT_PACKAGES_DIR}/lib/libgraphite2.la" "${CURRENT_PACKAGES_DIR}/debug/lib/libgraphite2.la")
 
-file(COPY "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
-file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/COPYING" "${SOURCE_PATH}/LICENSE")
 
-vcpkg_fixup_pkgconfig()

--- a/ports/graphite2/vcpkg.json
+++ b/ports/graphite2/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "graphite2",
   "version": "1.3.14",
-  "port-version": 3,
+  "port-version": 4,
   "description": [
     "Graphite is a \"smart font\" system developed specifically to handle the complexities of lesser-known languages of the world.",
     "Graphite2 is a rework of the original Graphite engine that is faster, smaller, and uses an API that is better suited to the layout architecture of most text-processing applications."

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3106,7 +3106,7 @@
     },
     "graphite2": {
       "baseline": "1.3.14",
-      "port-version": 3
+      "port-version": 4
     },
     "graphqlparser": {
       "baseline": "0.7.0",

--- a/versions/g-/graphite2.json
+++ b/versions/g-/graphite2.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "7d9a7404bf9dece906b2f6d8d8a70d7139d5e998",
+      "version": "1.3.14",
+      "port-version": 4
+    },
+    {
       "git-tree": "247c12c09e9766df6a90a8a45dad08320f4e0493",
       "version": "1.3.14",
       "port-version": 3

